__tests__/app.test.js

```diff
@@ -0,0 +1,52 @@
+const request = require("supertest")
+const app = require("../db/app")
+const db = require("../db/connection")
+const seed = require("../db/seeds/seed")
+const data = require("../db/data/test-data")
+
+beforeEach(() => {
+    return seed(data)
+})
+
+afterAll(() => {
+    return db.end()
+})
+
+describe('GET/API', () => {
+    test('200 OK: message', () => {
+        return request(app)
+        .get('/api')
+        .expect(200)
+        .then(({body}) => {
+            expect(body.msg).toBe("All OK")
+        })
+    })
+})
+
+describe('GET/API/TOPICS', () => {
+    test('200 OK: all topics', () => {
+        return request(app)
+        .get('/api/topics')
+        .expect(200)
+        .then(({body}) => {
+            const { topics } = body
+             expect(topics).toBeInstanceOf(Object)
+             expect(topics).toHaveLength(3)
+             topics.forEach((topic) => {
+                 expect(topic).toMatchObject({
+                     description: expect.any(String),
+                     slug: expect.any(String)
+                 })
+             })
+        })
+    })
+})
+
+describe('GET /api/topics (ERROR)', () => {
+    test('404 Not Found: invalid route', () => {
+      return request(app)
+        .get('/api/invalid')
+        .expect(404);
```

Comment:
I would expect a message to come back here and not just a 404 status. I think this is just coming back as a default from express at the minute.

We are going to cover this type of error in the lecture tomorrow, so you can come back to it then 😄